src/toil/deferred.py

```diff
@@ -318,19 +321,24 @@ def _runOrphanedDeferredFunctions(self):
                     continue
 
                 try:
-                    fcntl.lockf(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
-                except OSError:
-                    # File is still locked by someone else.
-                    # Look at the next file instead
-                    continue
+                    safe_lock(fd, block=False)
+                except OSError as e:
+                    if e.errno in (errno.EACCES, errno.EAGAIN):
+                        # File is still locked by someone else.
+                        # Look at the next file instead
+                        continue
+                    else:
+                        # Something else went wrong
+                        os.close(fd)
+                        raise
```

Comment:
Should we close the file in either case?  i.e.

```
                except OSError as e:
                    os.close(fd)
                    if e.errno in (errno.EACCES, errno.EAGAIN):
                        # File is still locked by someone else.
                        # Look at the next file instead
                        continue
                    else:
                        # Something else went wrong
                        raise
```

Reply:
Yeah, we do need a close before continuing.